src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -367,4 +370,78 @@ public boolean contains(String key) {
 
         return mPrefs.contains(key);
     }
+    
+    private TokenCacheItem getFamilyRefreshTokenItemForUser(String userId) {       
+        
+        Iterator<TokenCacheItem> results = this.getTokensForUser(userId).iterator();
+        TokenCacheItem tokenItem = new TokenCacheItem();   
+        while (results.hasNext()) {
+            final TokenCacheItem tokenCacheItem = results.next();
+            if(tokenCacheItem.getUserInfo() != null
+                    && !StringExtensions.IsNullOrBlank(tokenCacheItem.getFamilyClientId())
+                    && tokenCacheItem.getFamilyClientId().equalsIgnoreCase("1")){
+                tokenItem = tokenCacheItem;
+                //return the cacheItem which contains the FRT for this userId
+                //there should be at most only one FRT record for each userId
+                return tokenItem;
+            }
+        }
+        //If the cache does not contain the FRT for this user return null
+        Logger.i(TAG, "Cannot find the family refresh token cache item for this user in cache", "");
+        return null;
+    }
+    
+    protected String serialize(String uniqueUserId) throws UsageAuthenticationException {
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if((new BrokerProxy(mContext)).canSwitchToBroker()){
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
+        } else {
+            TokenCacheItem tokenItem = getFamilyRefreshTokenItemForUser(uniqueUserId);
+            
+            if(tokenItem == null) {
+                Logger.i(TAG, "Cannot serialize the FRT cache item: family refresh token is null", "");
+                return null;
+            }
+
+            Gson gson = new GsonBuilder()
+                    .registerTypeAdapter(BlobContainer.class, new BlobContainerAdapter())
```

Comment:
Well, as far as I know this is the best way I know if we need to compare the version UID explicitly and store it in the json. 

However, I do not think we have the need to serialize the version UID. Because Gson will compare the compatibility of the input object with target class automatically. I worte a test demo for the serialize/deserialize in gson. I made some change of the target class. And I found out that the change of versionUID did not have any impact on the deserialize process. As long as the attributes of the input object and target class are compatible, where they are parent-child classes, or let's say they have an intersection with the attrinutes, the  input object can be deserialized into the target class. If the input object and target class are not compatible, the fromGson() function will throw the IllegalStateException, which is as expected here. 

To sum up, I am quite doubtful about the need to serialized the versionUID if we use Gson. @weijjia  @RandalliLama  @RPangrle